datafusion/proto/src/bytes/mod.rs

```diff
@@ -332,7 +361,8 @@ mod test {
                 println!("testing: {n}");
 
                 let expr_base = col("a").lt(lit(5i32));
-                let expr = (0..n).fold(expr_base.clone(), |expr, _| expr.and(expr_base.clone()));
+                // Generate a tree of AND and OR expressions (no subsequent ANDs or ORs).
```

Comment:
I would love to see a test that made the other type of nesting `(expr_base.clone().and(expr))`

Reply:
Thanks, added one (`roundtrip_deeply_nested_binary_expr_reverse_order`).